core-contracts/tests/bns/name_register_test.ts

```diff
@@ -0,0 +1,565 @@
+                [
+                    types.buff(ripemd160),
+                    types.uint(2560000),
+                ], 
+                bob.address),
+        ]);
+        assertEquals(block.height, 15);
+        block.receipts[0].result
+            .expectOk()
+            .expectUint(144 + block.height - 1);
+
+        // Bob registering the name 'bob.blockstack' should succeed
+        block = chain.mineBlock([
+            Tx.contractCall("bns", "name-register", 
+                [
+                    types.buff(cases[0].namespace),
+                    types.buff(name),
+                    types.buff(cases[0].salt),
+                    types.buff(cases[0].zonefile),
+                ], 
+                bob.address),
+        ]);
+        assertEquals(block.height, 16);
+        block.receipts[0].result
+            .expectOk()
+            .expectBool(true);
+
+        call = chain.callReadOnlyFn("bns", "resolve-principal", [types.principal(bob.address)], alice.address)
+        let response:any = call.result
+            .expectOk()
+            .expectTuple();
+        response["name"].expectBuff("bob");
+        response["namespace"].expectBuff("blockstack");
+
+        call = chain.callReadOnlyFn("bns", "name-resolve", [types.buff(cases[0].namespace), types.buff(name)], alice.address)
+        response = call.result
+            .expectOk()
+            .expectTuple();
+        response["owner"].expectPrincipal(bob.address);
+        response["zonefile-hash"].expectBuff(cases[0].zonefile);
+
+        // should fail registering twice
+        block = chain.mineBlock([
+            Tx.contractCall("bns", "name-register", 
+                [
+                    types.buff(cases[0].namespace),
+                    types.buff(name),
+                    types.buff(cases[0].salt),
+                    types.buff(cases[0].zonefile),
+                ], 
+                bob.address),
+        ]);
+        assertEquals(block.height, 17);
+        block.receipts[0].result
+            .expectErr()
+            .expectInt(2004);
+    
+        // Charlie registering 'bob.blockstack'
+        // should fail
+        name = "bob";
+        let salt = "1111"
+        merged = new TextEncoder().encode(`${name}.${cases[0].namespace}${salt}`);
+        sha256 = createHash("sha256")
+            .update(merged)
+            .digest();
+        ripemd160 = createHash("ripemd160")
+            .update(sha256)
+            .digest();
+        block = chain.mineBlock([
+            Tx.contractCall("bns", "name-preorder", 
+                [
+                    types.buff(ripemd160),
+                    types.uint(2560000),
+                ], 
+                charlie.address),
+        ]);
+        assertEquals(block.height, 18);
+        block.receipts[0].result
+            .expectOk()
+            .expectUint(144 + block.height - 1);
+
+        // Bob registering the name 'bob.blockstack' should succeed
```

Comment:
comment does not match test here either - charlie registering the name here fails